src/test/java/org/blyznytsia/context/AnnotationApplicationContextTest.java

```diff
@@ -52,4 +53,11 @@ void getAllBeans_givenValidType_shouldReturnBeans() {
     assertThat(beans.get("bean1").getClass()).isEqualTo(Bean1.class);
     assertThat(beans.get("bean2").getClass()).isEqualTo(Bean2.class);
   }
+
+  @Test
+  void getListBean_givenValidBeanNameAndTypeInList_shouldReturnBean() {
+    var bean7 = ctx.getBean("bean7", Bean7.class);
+    assertThat(bean7).isInstanceOf(Bean7.class);
```

Comment:
the test should check that an autowired list/collection parametrized with an interface type contains concrete implementations. 
Currently, it doesn't do much except create an empty list.